students/john_navitsky/session06/mailroom.py

```diff
@@ -168,13 +182,52 @@ def print_thank_you(donor_id,hint="wonderful",dest=sys.stdout):
     print_lines(2,dest)
     print("-"*80,file=dest)
 
-    print(letter.format(hint,**donors[donor_id]),file=dest)
+    print(letter.format(hint,**current_donor),file=dest)
 
     print("-"*80,file=dest)
     print_lines(2,dest)
 
 
-def thank_you_entry():
+def match_donor(current_donor, donors):
+        """ See if record matches existing record, if so, return full record. """
+        for donor_id in donors:
+            existing_donor=donors[donor_id]
+            if existing_donor["full_name"].lower() == current_donor["full_name"].lower():
+                # if the owner exists, pull in the complete donor record
+                current_donor=existing_donor
```

Comment:
this is a bit of odd logic -- you are always going to get a donor record back, but it may be the new one, or it may be an existing one -- maybe better to search the donor db for a name, and then you can return either the existing one, or a new one, rather than not being sure at this point which it is.

Reply:
At some point you're going to need to decide if a record is new or matches an existing record.  Personally, I'd prefer this to happen when they start to enter a record.  This particular function is a really a stub where it should be expanded to allow them to select an existing entry or determine it is a new donor.

Reply:
Yup — bit I think better to be more direct. 

And you don’t want to create incomplete records.